src/openapi-generator.ts

```diff
@@ -768,7 +773,10 @@ export class OpenAPIGenerator {
         );
       });
 
-      alreadyRequired = registeredSchema.required ?? [];
+      alreadyRequired =
+        'properties' in registeredSchema && registeredSchema.required
```

Comment:
Not as pretty, sorry 😢 but had to make the types happy

Reply:
I'm wondering if this highlights an issue. The `registeredSchema` can be a ReferenceObject, which would mean it can be `{ $ref: <some-third-schema> }` itself. @AGalabov do you think we'll break in this case? So two extends: `A.extend({ ... }).extend({ ... }`

Reply:
Tried to add such a test and it failed, but I'm not sure this is the reason - maybe if we fix that, then this logic would break?

![Screenshot 2022-10-14 at 0 28 04](https://user-images.githubusercontent.com/756418/195714260-dfdaa10a-b73d-4c54-a4a6-465f50eabe56.png)
<img width="518" alt="Screenshot 2022-10-14 at 0 29 40" src="https://user-images.githubusercontent.com/756418/195714454-16a310f0-5012-4000-b307-53601d7ac505.png">

Reply:
I'm a little confused here with your test. Is the extended schema 2 meant to extend schema 1?

Reply:
Yes, BaseSchema < ExtendedSchema < Extended2Schema, but later in the OpenAPI doc it's output as Base < Extended and Base < Extended2, so the Extended2 case is wrong

Reply:
Ah, I see what you mean, let me check again

Reply:
This is the real output if Extended2Schema extends ExtendedSchema

<img width="545" alt="Screenshot 2022-10-17 at 18 01 12" src="https://user-images.githubusercontent.com/756418/196212330-6fe75305-2e7b-4197-af6e-9e4bd01ce5f5.png">

Reply:
In any case, that's not a problem with your PR and we can solve it separately, I'll see if I can open a PR soon